### ISSUE TITLE
Changes to child task titles are not saved when using the button to create/open the parent

### ISSUE THREAD

**Issue by nekohayo** [getting-things-gnome/gtg#745]
1. Create (or open) a task
2. Type some additional words in the title of that task (not the body*)
3. Click the "Open parent" (or "Add parent") button

Result: the additional words you had typed in the title are lost.

 *: somehow, the body is not affected by this bug...

Rating the severity as critical because technically it can mean data loss for users, but low-hanging-fruit because it is most likely a trivial fix...

**Comment by zeddo123** [comment]
The first time I tried to reproduce it, I couldn't. but the next time it happened. So it's all about the delay in `light_save` in the editor. 

From what I've understood, since it hasn't been 7 seconds (`GnomeConfig.SAVETIME`) since the last save gtg won't save again which explains why the first time I didn't experience the bug (I probably left the window open long enough?)

**Comment by zeddo123** [comment]
What I haven't figured out yet after an hour of debugging is why it only affects the title. If I had to guess, I think that due to the fact that we're detect the title more often than the content, the delay is always under 7 seconds. @diegogangl, could that be the case?

**Comment by diegogangl** [comment]
If you launch with debugging enabled you can see how long the taskview takes to process everything. It's always a few ms, unless you have a lot of text.

This could be a race condition, or it could be something in the way liblarch queues changes (`set_parent()` comes from  `treenode.py` in liblarch).

**Comment by Neui** [comment]
I did some basic debugging and it seems it saves after opening the parent window (in the close/window destruction handler of the now child task), which includes setting the title. So i thought adding saving before setting it as the parent:

```diff
diff --git a/GTG/gtk/editor/editor.py b/GTG/gtk/editor/editor.py
index 5cde3131..108f03fb 100644
--- a/GTG/gtk/editor/editor.py
+++ b/GTG/gtk/editor/editor.py
@@ -793,6 +793,14 @@ class TaskEditor:
             parent = self.req.new_task(tags=tags, newtask=True)
             parent_id = parent.get_id()
 
+            # Copy-pasted from destruction()
+            if self.task.is_new():
+                tid = self.task.get_id()
+                self.req.delete_task(tid)
+            else:
+                self.save()
+                [sub.set_to_keep() for sub in self.task.get_subtasks() if sub]
+
             self.task.set_parent(parent_id)
             self.app.open_task(parent_id)
             # Prevent WM issues and risks of conflicting content changes:
```

This seems to work a little better, but if you click on the "Create parent" button while holding a key down (so that the title doesn't get to be in the title style), then the bug still occurs. I guess that the taskview processing needs to happen before closing/saving.

Also, the small change I did feels a bit too hacky, I'd expect the parent task to be automatically be updated with the new task name if the task name is being updated.
